File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 5/8/2026, 4:23:41 PM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        reference = strs[0]
        for i, char in enumerate(reference):
            for j in range(1,len(strs)):
                if i >= len(strs[j]) or char != strs[j][i]:
                    return reference[:i]
        
        return reference

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.c
// Last updated: 5/8/2026, 4:23:48 PM
char* longestCommonPrefix(char** strs, int strsSize) {
    if(strsSize == 0)
    {
        static char ans[] = "";
        return ans;
    }

    char* prefix = strs[0];
    int prefixlen = strlen(prefix);

    for(int i=0; i<strsSize; i++)
    {
        while(strncmp(prefix,strs[i],prefixlen)!= 0)
        {
            prefixlen--;
            prefix[prefixlen] = '\0';
        }
        if(prefixlen ==0)
            return prefix;
        
    }
    return prefix;
}